core/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java

```diff
@@ -39,15 +38,12 @@
 	@Autowired
 	private AnalysisEngineCacheService cacheService;
 	@Autowired
-	private UserService userService;
-	@Autowired
 	private UserAnonymizedService userAnonymizedService;
 	@Autowired
 	private MessageService messageService;
 
-	public void analyze(UUID userID, AppActivityDTO[] appActivities)
+	public void analyze(UUID userAnonymizedID, AppActivityDTO[] appActivities)
 	{
-		UUID userAnonymizedID = userService.getPrivateUser(userID).getPrivateData().getUserAnonymizedID();
 		UserAnonymizedDTO userAnonymized = userAnonymizedService.getUserAnonymized(userAnonymizedID);
 		for (AppActivityDTO appActivity : appActivities)
```

Comment:
The logic below assumes that the activities are added in order, oldest first. Instead of relying on the client to send them in that order, I propose to sort them here.